### ISSUE TITLE
Configurable JSON (or any one-line format) logging

### ISSUE THREAD

**Issue by tommasopozzetti** [projectcalico/calico#3534]
## Expected Behavior
Always produce logs in a one-line format to ease the process of ingesting and processing them.

## Current Behavior
I've seen this behavior in the calico cni-plugin. With log level set to info, the plugin logs (through the kubelet) pods and namespaces structures that are worked on (see examples [here](https://github.com/projectcalico/cni-plugin/blob/b1b643a7b7d21e1a3597b953df9651124269660f/pkg/k8s/k8s.go#L778) and [here](https://github.com/projectcalico/cni-plugin/blob/b1b643a7b7d21e1a3597b953df9651124269660f/pkg/k8s/k8s.go#L787)). Because of logging with `%v`, if any string in the structure contains a newline character, the log is split on multiple lines. A simple example is the case in which a namespace is created via `kubectl apply`. This utility adds an annotation by default which terminates with a `\n`.

This behavior makes it very hard to process logs and properly run them through ingestion pipelines.

## Possible Solution
Make the logging format configurable [here](https://github.com/projectcalico/cni-plugin/blob/b1b643a7b7d21e1a3597b953df9651124269660f/internal/pkg/utils/utils.go#L575) so that JSON formatting can be selected (or any format that will properly escape newlines in strings).

**Comment by caseydavenport** [comment]
Yes, seems reasonable to support an alternate log format that is easier to ingest. I think this might go hand-in-hand with some other ideas that have been floating around to switch off of logrus for logging.

I'd like to see an option to use json formatted logs for ingestion.

**Comment by AshokMishra** [comment]
Seems reasonable. Any updates?
